src/main/java/springboot/test/repository/UserRepository.java

```diff
@@ -0,0 +1,9 @@
+package springboot.test.repository;
+
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import springboot.test.entity.User;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    Optional<User> getFirstByPhoneNumber(String number);
```

Comment:
see checklist - we should fetch roles along with user in this method as well as in `getById()`